sources/src/resources/init-scripts/gradle-actions.inject-develocity.init.gradle

```diff
@@ -90,75 +97,115 @@ if (gradleInjectionEnabled != "true") {
+                            buildScanExtension.captureTaskInputFiles = develocityCaptureFileFingerprints
                         }
                     }
-                    if (buildScan.metaClass.respondsTo(buildScan, 'setUploadInBackground', Boolean)) buildScan.uploadInBackground = buildScanUploadInBackground  // uploadInBackground not available for build-scan-plugin 1.16
-                    buildScan.value CI_AUTO_INJECTION_CUSTOM_VALUE_NAME, CI_AUTO_INJECTION_CUSTOM_VALUE_VALUE
+                    // uploadInBackground not available for build-scan-plugin 1.16
+                    if (buildScanExtension.metaClass.respondsTo(buildScanExtension, 'setUploadInBackground', Boolean)) buildScanExtension.uploadInBackground = buildScanUploadInBackground
+                    buildScanExtension.value CI_AUTO_INJECTION_CUSTOM_VALUE_NAME, CI_AUTO_INJECTION_CUSTOM_VALUE_VALUE
                 }
 
-                if (geUrl && geEnforceUrl) {
-                    pluginManager.withPlugin(BUILD_SCAN_PLUGIN_ID) {
-                        afterEvaluate {
-                            logger.lifecycle("Enforcing Develocity: $geUrl, allowUntrustedServer: $geAllowUntrustedServer, captureFileFingerprints: $geCaptureFileFingerprints")
-                            buildScan.server = geUrl
-                            buildScan.allowUntrustedServer = geAllowUntrustedServer
+                pluginManager.withPlugin(BUILD_SCAN_PLUGIN_ID) {
+                    afterEvaluate {
+                        if (develocityUrl && develocityEnforceUrl) {
+                            logger.lifecycle("Enforcing Develocity: $develocityUrl, allowUntrustedServer: $develocityAllowUntrustedServer for Build Scan plugin")
+                            buildScan.server = develocityUrl
+                            buildScan.allowUntrustedServer = develocityAllowUntrustedServer
+                        }
+
+                        if (buildScanTermsOfUseUrl && buildScanTermsOfUseAgree) {
+                            buildScan.termsOfServiceUrl = buildScanTermsOfUseUrl
+                            buildScan.termsOfServiceAgree = buildScanTermsOfUseAgree
                         }
-                        if (isAtLeast(gePluginVersion, '2.1') && atLeastGradle5) {
-                            if (isAtLeast(gePluginVersion, '3.7')) {
-                                buildScan.capture.taskInputFiles = geCaptureFileFingerprints
+
+                        if (isAtLeast(develocityPluginVersion, '2.1') && atLeastGradle5) {
+                            if (isAtLeast(develocityPluginVersion, '3.7')) {
+                                buildScan.capture.taskInputFiles = develocityCaptureFileFingerprints
                             } else {
-                                buildScan.captureTaskInputFiles = geCaptureFileFingerprints
+                                buildScan.captureTaskInputFiles = develocityCaptureFileFingerprints
                             }
                         }
```

Comment:
@welandaz and I discovered you cannot set the capture task input files flag after the project has been evaluated. [It's too late and results in an error](https://scans.gradle.com/s/hva3c63bl7kls/tests/task/:test/details/com.gradle.gradlebuildaction.TestDevelocityInjection/enforces%20Develocity%20URL%20and%20allowUntrustedServer%20in%20project%20if%20enforce%20url%20parameter%20is%20enabled%20%5BtestGradleVersion:%20Gradle%205.6.4%2C%20%232%5D?top-execution=1). https://github.com/gradle/actions/pull/58#discussion_r1514600865

Reply:
Thank you for pointing it out. Restored in https://github.com/gradle/actions/pull/62/commits/d19d801ed6180a8918df48c156960ff3c8519a4a